roles/ceph-osd/tasks/pool_names.yml

```diff
@@ -0,0 +1,144 @@
+  shell: rados lspools
+  register: existing_pools
+
+# if default pool doesn't exist
+- name: set pool names if default pool doesn't exist
+  set_fact:
+    ssd_pool: "rbd_ssd"
+    hybrid_pool: "rbd_hybrid"
+  when: '"default" not in existing_pools.stdout'
+
+# if default pool exist and ssd/hybrid pool doesn't exist
+- name: get first osd type
+  shell: ceph osd tree |sort -rn -k1 |grep -m1  -oP "(?<=host )\w+"
+  register: result_first_osd_host
+  when: ("default" in existing_pools.stdout) and
+        "rbd_ssd" not in existing_pools.stdout and
+        "rbd_hybrid" not in existing_pools.stdout
+
+- name: set pool name=default
+  set_fact:
+    ssd_pool: "default"
+    hybrid_pool: "rbd_hybrid"
+  when: result_first_osd_host is defined and
+        result_first_osd_host.stdout in groups["ceph_osds_ssd"]
+
+- name: set pool name=default
+  set_fact:
+    hybrid_pool: "default"
+    ssd_pool: "rbd_ssd"
+  when: result_first_osd_host is defined and
+        result_first_osd_host.stdout in groups["ceph_osds_hybrid"]
+
+# if default pool exists, ssd/hybrid pool exists too
+- name: set pool names
+  set_fact:
+    ssd_pool:  "rbd_ssd"
+    hybrid_pool: "default"
+  when: ("default" in existing_pools.stdout) and
+        "rbd_ssd" in existing_pools.stdout
+
+- name: set pool names
+  set_fact:
+    hybrid_pool:  "rbd_hybrid"
+    ssd_pool: "default"
+  when: ("default" in existing_pools.stdout) and
+        "rbd_hybrid" in existing_pools.stdout
+
+- assert:
+    that:
+      - ssd_pool is defined
+      - hybrid_pool is defined
+
+- name: populate pool names
+  set_fact:
+    hybrid_pool:  "{{ hybrid_pool }}"
+    ssd_pool: "{{ ssd_pool }}"
+  delegate_to: "{{ item }}"
+  delegate_facts: yes
+  with_items: "{{ groups['ceph_osds'] }}"
+
+# configure ruleset names
+- name: initial ruleset names
+  set_fact:
+    ssd_ruleset: "ssd_ruleset"
+    hybrid_ruleset: "hybrid_ruleset"
+
+- name: set ruleset name for default pool
+  set_fact:
+    ssd_ruleset: "replicated_ruleset"
+  when: ssd_pool == "default"
+
+- name: set ruleset name for default pool
+  set_fact:
+    hybrid_ruleset: "replicated_ruleset"
+  when: hybrid_pool == "default"
+
+- name: populate ruleset to all osd nodes
+  set_fact:
+    ssd_ruleset: "{{ ssd_ruleset }}"
+    hybrid_ruleset: "{{ hybrid_ruleset }}"
+  delegate_to: "{{ item }}"
+  delegate_facts: yes
+  with_items: "{{ groups['ceph_osds'] }}"
+
+- name: set dict of ceph pool
+  set_fact:
+    ceph_pools: { "rbd_ssd": { "pool_name": "{{ ssd_pool }}", "enabled": "{{ groups['ceph_osds_ssd'] |default([]) |length > 0 }}", "volume_type": "CEPH_SSD" },
+                  "rbd_hybrid": { "pool_name": "{{ hybrid_pool }}", "enabled": "{{ groups['ceph_osds_hybrid'] |default([]) |length > 0 }}", "volume_type": "CEPH_HYBRID" } }
+  delegate_to: "{{ item }}"
+  delegate_facts: yes
+  with_items:
+    - "{{ groups['controller'] }}"
+    - "{{ groups['cinder_volume'] }}"
+    - "{{ inventory_hostname }}"
```

Comment:
Yes, we have. it worked as expected.
As controller1,2 belong to both groups['controller'], groups['cinder_volume'],
so it will executed twice on controller1,2.
But it doesn't matter, the result is the same.